fix clang 3.7.1 warnings

warnings building mantid with clang 3.7.1.
```
mac83961:build2 svh$ ninja
[568/2948] Building CXX object Framework/PythonInterface/mantid/api/CMakeFiles/PythonAPIModule.dir/src/Exports/DataProcessorAlgorithm.cpp.o
In file included from /Users/svh/Documents/MantidProject/mantid/Framework/PythonInterface/mantid/api/src/Exports/DataProcessorAlgorithm.cpp:1:
/Users/svh/Documents/MantidProject/mantid/Framework/PythonInterface/inc/MantidPythonInterface/api/PythonAlgorithm/DataProcessorAdapter.h:86:60: warning: all paths through this function will call itself [-Winfinite-recursion]
  void setPropManagerPropName(const std::string &propName) {
                                                           ^
/Users/svh/Documents/MantidProject/mantid/Framework/PythonInterface/inc/MantidPythonInterface/api/PythonAlgorithm/DataProcessorAdapter.h:91:62: warning: all paths through this function will call itself [-Winfinite-recursion]
                       const std::string &nameInPropManager) {
                                                             ^
2 warnings generated.
[638/2948] Building CXX object Framework/PythonInterface/mantid/api/CMakeFiles/PythonAPIModule.dir/src/PythonAlgorithm/DataProcessorAdapter.cpp.o
In file included from /Users/svh/Documents/MantidProject/mantid/Framework/PythonInterface/mantid/api/src/PythonAlgorithm/DataProcessorAdapter.cpp:1:
/Users/svh/Documents/MantidProject/mantid/Framework/PythonInterface/inc/MantidPythonInterface/api/PythonAlgorithm/DataProcessorAdapter.h:86:60: warning: all paths through this function will call itself [-Winfinite-recursion]
  void setPropManagerPropName(const std::string &propName) {
                                                           ^
/Users/svh/Documents/MantidProject/mantid/Framework/PythonInterface/inc/MantidPythonInterface/api/PythonAlgorithm/DataProcessorAdapter.h:91:62: warning: all paths through this function will call itself [-Winfinite-recursion]
                       const std::string &nameInPropManager) {
                                                             ^
2 warnings generated.
[1242/2948] Building CXX object Framework/CurveFitting/CMakeFiles/CurveFitting.dir/src/Functions/ComptonProfile.cpp.o
/Users/svh/Documents/MantidProject/mantid/Framework/CurveFitting/src/Functions/ComptonProfile.cpp:21:14: warning: unused variable 'STDDEV_TO_HWHM' [-Wunused-const-variable]
const double STDDEV_TO_HWHM = std::sqrt(std::log(4.0));
             ^
1 warning generated.
[2948/2948] Linking CXX executable bin/MantidPlot.app/Contents/MacOS/MantidPlot
```

--- a/Framework/CurveFitting/src/Functions/ComptonProfile.cpp
+++ b/Framework/CurveFitting/src/Functions/ComptonProfile.cpp
@@ -17,8 +17,6 @@ namespace {
 ///@cond
 // const char * WSINDEX_NAME = "WorkspaceIndex";
 const char *MASS_NAME = "Mass";
-
-const double STDDEV_TO_HWHM = std::sqrt(std::log(4.0));
 ///@endcond
 }
 

--- a/Framework/PythonInterface/inc/MantidPythonInterface/api/PythonAlgorithm/DataProcessorAdapter.h
+++ b/Framework/PythonInterface/inc/MantidPythonInterface/api/PythonAlgorithm/DataProcessorAdapter.h
@@ -83,19 +83,12 @@ public:
       const std::string &propertyManager = std::string()) {
     return this->getProcessProperties(propertyManager);
   }
-  void setPropManagerPropName(const std::string &propName) {
-    this->setPropManagerPropName(propName);
-  }
-
-  void mapPropertyName(const std::string &nameInProp,
-                       const std::string &nameInPropManager) {
-    this->mapPropertyName(nameInProp, nameInPropManager);
-  }
 
   API::Workspace_sptr assembleProxy(const std::string &partialWSName,
                                     const std::string &outputWSName) {
     return this->assemble(partialWSName, outputWSName);
   }
+
   void saveNexusProxy(const std::string &outputWSName,
                       const std::string &outputFile) {
     this->saveNexus(outputWSName, outputFile);

